python/revng/cli/_commands/fetch_debuginfo/elf.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python3
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import os
+import shutil
+import subprocess
+from pathlib import Path
+
+from elftools.elf.elffile import ELFFile
+from elftools.elf.sections import NoteSection
+
+from .common import download_file, log
+
+
+def parse_build_id(file):
+    for section in file.iter_sections():
+        if not isinstance(section, NoteSection):
+            continue
+        for note in section.iter_notes():
+            desc = note["n_desc"]
+            if note["n_type"] != "NT_GNU_BUILD_ID":
+                continue
+            return desc
+    return None
+
+
+def parse_debug_link(data):
+    dataptr = 0
+    while dataptr < len(data):
+        while dataptr < len(data) and not (32 <= data[dataptr] <= 127):
+            dataptr += 1
+
+        if dataptr >= len(data):
+            break
+
+        endptr = dataptr
+        while endptr < len(data) and data[endptr] != 0:
+            endptr += 1
+
+        # TODO: If we want to parse checksum as well, we should continue
+        # and set up `dataptr = endptr`.
+        return data[dataptr:endptr].decode("latin-1")
+
+    return None
+
+
+def check_dbgsym_packages(verbose):
+    global_debug_package_list_path = "/etc/apt/sources.list.d/ddebs.list"
+    if os.path.exists(global_debug_package_list_path):
+        log("dbgsym packages are enabled...", verbose)
+        return True
+
+    log(
+        "warning: Please enable dbgsym packages by following "
+        "https://wiki.ubuntu.com/Debug%20Symbol%20Packages",
+        verbose,
+    )
```

Comment:
This is Ubuntu-specific, also the user might have called the file something different from `ddebs.list`.

I think fastest way to check this and be compatible with both Ubuntu and Debian is to check if any repos has `-debug` suffix (which seems to be the [Debian's way](https://wiki.debian.org/AutomaticDebugPackages)) or the repo URL contains `ddebs.ubuntu.com` (which seems to be [Ubuntu's way](https://wiki.ubuntu.com/Debug%20Symbol%20Packages)).

We need to do this for `/etc/apt/sources.list` and for each file in `/etc/apt/sources.list.d/`, considering each line starting with `deb`.

The alternative would be `apt-cache search '\-dbgsyms' | grep -- '-dbgsyms '` but that's soooo slow.

Reply:
Ah....yes.... I actually don't like either way of these two. Can we just do it unconditionally (without checking if ```check_dbgsym_packages``` is ```true```)? Just be specific in the message that it will work if the ```service``` was enabled?

Reply:
* What `service`?
* So you want to simply run `dpkg` (check if it's available)? Works!

Reply:
Yeah, exactly that, without making the hacking/observation of users ```sources.list```.